drivers/fpga/dfl-cxl-cache.c

```diff
@@ -263,6 +248,74 @@ static int cxl_cache_dma_region_add(struct dfl_cxl_cache *cxl_cache,
 	return 0;
 }
 
+static void fixup_ptes(struct mm_struct *mm, unsigned long start, unsigned long end)
+{
+	unsigned long addr;
+	pgd_t *pgd;
+	p4d_t *p4d;
+	pud_t *pud;
+	pmd_t *pmd;
+	pte_t *pte;
+
+	for (addr = start; addr < end; addr += PAGE_SIZE) {
+		pgd = pgd_offset(mm, addr);
+		if (pgd_bad(*pgd) || pgd_none(*pgd))
+			continue;
+
+		p4d = p4d_offset(pgd, addr);
+		if (p4d_bad(*p4d) || p4d_none(*p4d))
+			continue;
+
+		pud = pud_offset(p4d, addr);
+		if (pud_bad(*pud) || pud_none(*pud))
+			continue;
+
+		pmd = pmd_offset(pud, addr);
+		if (pmd_bad(*pmd) || pmd_none(*pmd))
+			continue;
+
+		pte = pte_offset_kernel(pmd, addr);
+		if (!pte_none(*pte) && pte_present(*pte))
+			*pte = pte_wrprotect(*pte);
+	}
+}
+
+static long cxl_cache_set_region_read_only(struct dfl_cxl_cache *cxl_cache,
+					   struct dfl_cxl_cache_buffer_region *region)
+{
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 13, 0)
```

Comment:
I believe it compiled on RHEL 8.6 and up. Would it be functional?
```suggestion
#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 13, 0) && RHEL_RELEASE_CODE < 0x806
```

Reply:
I can try your suggestion, but to be frank we've only ever tested dfl-cxl-cache.c in 6.1-lts kernel.

Reply:
RHEL ≤ 8.5 are pretty much [EOL](https://access.redhat.com/support/policy/updates/errata) at this point (apart from Updated Service for SAP Solutions), so it would be nice to drop this `#if` altogether at some point (see the relatively recent https://github.com/OFS/linux-dfl-backport/pull/107 though).